src/d3d12/d3d12_descriptor_heap.cpp

```diff
@@ -14,22 +14,31 @@ namespace wr::d3d12
 
 		heap->m_create_info = descriptor;
 		heap->m_increment_size = n_device->GetDescriptorHandleIncrementSize(static_cast<D3D12_DESCRIPTOR_HEAP_TYPE>(descriptor.m_type));
+		heap->m_native.resize(descriptor.m_versions);
 
-		D3D12_DESCRIPTOR_HEAP_DESC heap_desc;
-		heap_desc.NumDescriptors = descriptor.m_num_descriptors;
-		heap_desc.Type = (D3D12_DESCRIPTOR_HEAP_TYPE)descriptor.m_type;
-		heap_desc.Flags = descriptor.m_shader_visible ? D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE : D3D12_DESCRIPTOR_HEAP_FLAG_NONE;
-		heap_desc.NodeMask = 0;
-		HRESULT hr = n_device->CreateDescriptorHeap(&heap_desc, IID_PPV_ARGS(&heap->m_native));
+		for (uint32_t i = 0; i < descriptor.m_versions; ++i)
+		{
+
+			D3D12_DESCRIPTOR_HEAP_DESC heap_desc;
+			heap_desc.NumDescriptors = descriptor.m_num_descriptors;
+			heap_desc.Type = (D3D12_DESCRIPTOR_HEAP_TYPE)descriptor.m_type;
+			heap_desc.Flags = descriptor.m_shader_visible ? D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE : D3D12_DESCRIPTOR_HEAP_FLAG_NONE;
+			heap_desc.NodeMask = 0;
+			HRESULT hr = n_device->CreateDescriptorHeap(&heap_desc, IID_PPV_ARGS(&heap->m_native[i]));
```

Comment:
We have a macro that does the HRESULT for you and logs a message when it fails.. Its called `TRY_M` `TRY_M(result, "mymessage")`